### PR TITLE
Add search service

### PR DESCRIPTION
**Pull request by derwehr**
This PR adds a service using the backend's search API.

### CODE REVIEW

**Comment by derwehr** [comment]
only relevant files are
`frontend/src/services/QueryService.ts`, `frontend/src/services/QueryServiceImpl.ts` and `frontend/src/services/FakeDataQueryService.ts`
The rest is from merging the http-service into this branch

File: frontend/src/services/QueryService.ts [hunk 1]
```diff
@@ -19,4 +20,19 @@ export default abstract class QueryService {
...
+  /**
+   * Searches through all entities and entity types and returns those, that
+   * match the searchString. Only returns those values with the prefix of searchStrings.
+   * Values with n spaces are considered as n+1 single values.
+   * @param searchString The string to search for.
+   * @param path the path of the search API describes what to search for, can be 'all' 'nodes' or edges'.
+   * @param cancellation A CancellationToken used to cancel the asynchronous operation.
+   * @returns A promise that when evaluated, contains a {@link SearchResult} (nodes, edges, edgetypes and nodetypes).
+   */
+  public abstract fullTextSearch(
```
Comment (CatoLeanTruetschel) [inline frontend/src/services/QueryService.ts:33]: Why isn't this in a dedicated service? In the backend it is not part of this service and the HTTP API is on the subpath `search/` while all urls this service maes available is on the root path.

File: frontend/src/services/QueryServiceImpl.ts [hunk 2]
```diff
@@ -1,94 +1,48 @@
...
-      request.onabort = () => {
-        cancellationUnsubscribe();
-        reject(new CancellationError());
-      };
+    return this.http.post<QueryResult>(url, query, cancellation);
+  }
 
-      request.send(JSON.stringify(query ?? {}));
-    });
+  public fullTextSearch(
```
Comment (CatoLeanTruetschel) [inline frontend/src/services/QueryServiceImpl.ts:29]: Why isn't this in a dedicated service? In the backend it is not part of this service and the HTTP API is on the subpath `search/` while all urls this service maes available is on the root path.

File: frontend/src/services/QueryServiceImpl.ts [hunk 3]
```diff
@@ -1,94 +1,48 @@
...
+    searchString: string,
+    path: string,
+    cancellation?: CancellationToken
+  ): Promise<SearchResult> {
+    const possiblePaths = ['all', 'nodes', 'edges'];
```
Comment (CatoLeanTruetschel) [inline frontend/src/services/QueryServiceImpl.ts:34]: This is bad design. Add three different services to the method `searchAll()`, `searchNodes()`, `searchEdges()`. At least the typescript type annotation for the path parameter should be `'all' | 'nodes' | 'edges'`.
The backend does not support isolated search for nodes and edges any longer.

**Review by CatoLeanTruetschel — CHANGES_REQUESTED** [review]

File: frontend/src/services/QueryService.ts (hunk 1 above)
Comment (derwehr) [reply frontend/src/services/QueryService.ts]: Thanks for the quick review! Moved this to a dedicated service in ea9e726

File: frontend/src/services/QueryServiceImpl.ts (hunk 2 above)
Comment (derwehr) [reply frontend/src/services/QueryServiceImpl.ts]: See comment above

File: frontend/src/services/QueryServiceImpl.ts (hunk 3 above)
Comment (derwehr) [reply frontend/src/services/QueryServiceImpl.ts]: Changed this to only GET `/search/all` in ea9e726

**Review by joluj — APPROVED** [review]

**Review by CatoLeanTruetschel — APPROVED** [review]

**Pull request merged**